File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void dfs(TreeNode* node, vector<int> &inorder) {
    if (node == NULL) return;
    dfs(node->left, inorder);
    inorder.push_back(node->val);
    dfs(node->right, inorder);
}
    
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> v;
        dfs(root, v);
        return v;
    }
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */


const inorder = (root, ans ) => {
    if(root === null) return;
    inorder(root.left, ans);
    ans.push(root.val);
    inorder(root.right, ans);
}

var inorderTraversal = function(root) {
    const ans = [];
    inorder(root, ans);
    return ans;
};